packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -0,0 +1,834 @@
+   * @returns A promise that resolves to the success of the operation.
+   */
+  async addNewSeedPhraseBackup(
+    seedPhrase: Uint8Array,
+    password: string, // TODO: to verify whether we need the password here, check how multi-srp is handled in the keyring first.
+  ): Promise<void> {
+    // verify the password and unlock the vault
+    const { toprfEncryptionKey, toprfAuthKeyPair } =
+      await this.#verifyPassword(password);
+
+    // encrypt and store the seed phrase backup
+    await this.#encryptAndStoreSeedPhraseBackup(
+      seedPhrase,
+      toprfEncryptionKey,
+      toprfAuthKeyPair,
+    );
+  }
+
+  /**
+   * Add array of new seed phrase backups to the metadata store in batch.
+   *
+   * @param seedPhrases - The seed phrases to backup.
+   * @param password - The password used to create new wallet and seedphrase
+   * @returns A promise that resolves to the success of the operation.
+   */
+  async batchAddSeedPhraseBackups(seedPhrases: Uint8Array[], password: string) {
+    const { toprfEncryptionKey, toprfAuthKeyPair } =
+      await this.#verifyPassword(password);
+
+    // prepare seed phrase metadata
+    const seedPhraseMetadataArr =
+      SeedPhraseMetadata.fromBatchSeedPhrases(seedPhrases);
+
+    try {
+      // encrypt and store the seed phrase backups
+      await this.#withPersistedSeedPhraseBackupsState(async () => {
+        await this.toprfClient.batchAddSecretDataItems({
+          secretData: seedPhraseMetadataArr.map((metadata) =>
+            metadata.toBytes(),
+          ),
+          encKey: toprfEncryptionKey,
+          authKeyPair: toprfAuthKeyPair,
+        });
+        return seedPhrases;
+      });
+    } catch (error) {
+      log('Error encrypting and storing seed phrase backups', error);
+      throw new Error(
+        SeedlessOnboardingControllerError.FailedToEncryptAndStoreSeedPhraseBackup,
+      );
+    }
+  }
+
+  /**
+   * @description Fetch seed phrase metadata from the metadata store.
+   * @param params - The parameters for fetching seed phrase metadata.
+   * @param params.authConnectionId - OAuth authConnectionId from dashboard
+   * @param params.groupedAuthConnectionId - Optional grouped authConnectionId to be used for the authenticate request.
+   * You can pass this to use aggregate connection.
+   * @param params.userId - user email or id from Social login
+   * @param params.password - The password used to create new wallet and seedphrase
+   * @returns A promise that resolves to the seed phrase metadata.
+   */
+  async fetchAllSeedPhrases(params: {
+    authConnectionId: string;
+    groupedAuthConnectionId?: string;
+    userId: string;
+    password: string;
+  }): Promise<Uint8Array[]> {
+    const { authConnectionId, groupedAuthConnectionId, userId, password } =
+      params;
+    this.#assertIsValidNodeAuthTokens(this.state.nodeAuthTokens);
+
+    const { encKey, authKeyPair } = await this.#recoverEncKey({
```

Comment:
is this the function used to rehydrate/relogin user account, we should simply name it `rehydrate`

Reply:
i think `recoverEncKey` is fine, as it describes clearly what the function does.
`rehydrate` is not as explicit. it could mean many things.

Reply:
ah srry, i meant to change `fetchAllSeedPhrases` to `rehydrate` or `rehydrateAccounts`, because this `fetchAllSeedPhrases` is called when existing user logs in on a new device.